### PR TITLE
br/stream: added truncate log

### LINKED ISSUE #33091: br/stream: Support truncate unnecessary files via CLI
## Enhancement

This requires a new subcommand of `br stream` -- `truncate --until <ts-string>`. Which:

- Remove all log files contains only data before <ts-string>.
- Update the metafiles involved.

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yu745514916@live.com>

### What problem does this PR solve?

Issue Number: close #33091 

Problem Summary:
See the issue.

### What is changed and how it works?

This PR added a new subcommand for `stream`, which would truncate stale log files.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code
image

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/local.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func (l *LocalStorage) WalkDir(ctx context.Context, opt *WalkOption, fn func(str
 		// so use Rel to convert to relative path to l.base
 		path, _ = filepath.Rel(l.base, path)
```
Comment (3pointer) [inline br/pkg/storage/local.go:78]: so this line can be removed

File: br/pkg/restore/stream_metas.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	}
+}
+
+// OverFilesFullyBefore runs the function over all files contian data before the timestamp only.
+//   0                                          before
+//   |------------------------------------------|
+//    |-file1---------------| <- File contains records in this TS range would be found.
+//                                  |-file2--------------| <- File contains any record out of this won't be found.
+// This function would call the `f` over file1 only.
+func (ms *StreamMetadataSet) OverFilesFullyBefore(before uint64, f func(d *backuppb.DataFileInfo) (shouldBreak bool)) {
```
Comment (3pointer) [inline br/pkg/restore/stream_metas.go:56]: may over -> iterate or find 🤔

File: br/pkg/task/stream.go [hunk 3]
```diff
@@ -52,6 +53,9 @@ import (
 )
 
 const (
+	flagYes                     = "yes"
+	flagDryRun                  = "dry-run"
+	flagUntil                   = "until"
```
Comment (3pointer) [inline br/pkg/task/stream.go:58]: should ask PM confirm too...

**Review by 3pointer — COMMENTED** [review]
rest LGTM

File: br/cmd/br/stream.go [hunk 4]
```diff
@@ -139,6 +140,19 @@ func newStreamRestoreCommand() *cobra.Command {
 	return command
 }
 
+func newStreamTruncateCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "truncate",
```
[the remark this answers is not in the document]
Comment (YuJuncen) [reply br/cmd/br/stream.go:145]: yes.

File: br/pkg/storage/local.go (hunk 1 above)
Comment (YuJuncen) [reply br/pkg/storage/local.go:78]: I guess we'd better revert this change for keep the behavior consistent between `local` and `s3`.🤔

File: br/pkg/task/stream.go (hunk 3 above)
Comment (YuJuncen) [reply br/pkg/task/stream.go:58]: cc @IANTHEREAL 
(`--yes` and `--dry-run` may be hidden?)

**Review by 3pointer — APPROVED** [review]

File: br/pkg/task/stream.go (hunk 3 above)
Comment (YuJuncen) [reply br/pkg/task/stream.go:58]: checked

**Review by joccau — COMMENTED** [review]
rest LGTM

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**